sdk-lib/src/androidTest/java/cash/z/ecc/android/sdk/util/TestWallet.kt

```diff
@@ -79,9 +77,9 @@ class TestWallet(
 
     val available get() = synchronizer.saplingBalances.value?.available
     val unifiedAddress =
-        runBlocking { DerivationTool.deriveUnifiedAddress(seed, network = network, Account.DEFAULT) }
+        runBlocking { synchronizer.getCurrentAddress(Account.DEFAULT) }
     val transparentAddress =
-        runBlocking { DerivationTool.deriveTransparentAddress(seed, network = network, Account.DEFAULT) }
+        runBlocking { synchronizer.getLegacyTransparentAddress(Account.DEFAULT) }
```

Comment:
You shouldn't use the legacy account. We don't use it neither expose it anymore. 

Both SDKs should be consistent

Reply:
That is not what this API does. It returns the transparent receiver from the current UA for the given account. "Legacy" here just means "you should be using UAs primarily in your UI, but here's a taddr if you need it".

Reply:
This is misleading. Transparent addresses are not being called legacy anywhere else that I know of. We are referring to  components of UAs as receivers. Mixing deprecation intentions with these APIs is weird

Reply:
"Receiver" is a term of art that I thought would be more confusing in the mobile SDKs as it is a new term, and given that these APIs were intended to be direct swaps in for existing "show transparent address" usage introduced by the auto shielding feature, this name would be more obvious. But if you and @Carter agree that it should be exposed to your users as receivers, and your migration documentation will handle the onboarding, then it doesn't bother me.

Reply:
I think our best option is to have a strongly typed UnifiedAddress object that allows fetching the various components from it.  However I'd rather avoid further changing the API at this stage so that we can get this SDK update released.  We have enough churn right now as-is.

To track the future API change we can repurpose this existing issue, although the description needs to change a bit since it was written a while ago: https://github.com/zcash/zcash-android-wallet-sdk/issues/483